staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go

```diff
@@ -674,6 +674,11 @@ func kmsPrefixTransformer(ctx context.Context, config *apiserverconfig.KMSConfig
 	kmsName := config.Name
 	switch config.APIVersion {
 	case kmsAPIVersionV1:
+		if utilfeature.DefaultFeatureGate.Enabled(features.RemoveKMSv1) {
+			return storagevalue.PrefixTransformer{}, nil, nil, fmt.Errorf("KMSv1 is deprecated with no further fixes and will be removed in a future release. Use KMSv2 instead")
+		}
+		klog.Warning("KMSv1 is deprecated with no further fixes and will be removed in a future release. Use KMSv2 instead")
```

Comment:
Removal comment is too forward looking.

```suggestion
		klog.Warning("KMSv1 is deprecated and will only receive security updates going forward. Use KMSv2 instead.")
```